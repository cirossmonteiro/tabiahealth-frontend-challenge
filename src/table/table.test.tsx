import {act} from 'react';
import {render, screen} from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

import Table from './';

test('renders learn react link', async () => {
  const component = (
    <Table
      categories={[{ name: "category", icon: ["solid", "house"] }]}
      countries={["country"]}
      data={[[10]]}
    />
  );

  render(component);

  // testing sorting icons
  expect(screen.getByTestId('sort-icon').className).toBe("fa-solid fa-sort");
  userEvent.click(screen.getByTestId('sort-icon'));
  expect(screen.getByTestId('sort-icon').className).toBe("fa-solid fa-sort-up");
  userEvent.click(screen.getByTestId('sort-icon'));
  expect(screen.getByTestId('sort-icon').className).toBe("fa-solid fa-sort-down");
  userEvent.click(screen.getByTestId('sort-icon'));
  expect(screen.getByTestId('sort-icon').className).toBe("fa-solid fa-sort");

  const rendered = renderer.create(component);
  let tree = rendered.toJSON();
  expect(tree).toMatchSnapshot();
});
