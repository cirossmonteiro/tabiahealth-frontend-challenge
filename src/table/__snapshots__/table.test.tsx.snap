// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  className="main d-flex"
>
  <table>
    <thead>
      <tr>
        <th
          className="d-flex justify-content-center teams-sort"
        >
          <div
            className="d-flex"
          >
            Teams
            <div
              className="ms-1"
            >
              <i
                className="fa-solid fa-sort"
                data-testid="sort-icon"
                onClick={[Function]}
              />
            </div>
          </div>
        </th>
        <th
          className="category"
        >
          <div
            className="holder2 d-flex justify-content-center align-items-center"
          >
            <i
              className="mb-2 fa-solid fa-house"
            />
            category
          </div>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          className="country"
        >
          <div
            className="p-2 d-flex justify-content-center align-items-center"
          >
            All teams
          </div>
        </th>
        <td>
          <div
            className="sc-braxZu iumnsd d-flex justify-content-center align-items-center"
            value={10}
          >
            10.0
          </div>
        </td>
      </tr>
      <tr>
        <th
          className="country"
        >
          <div
            className="p-2 d-flex justify-content-center align-items-center"
          >
            country
          </div>
        </th>
        <td>
          <div
            className="sc-braxZu iumnsd d-flex justify-content-center align-items-center"
            value={10}
          >
            10.0
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
